### ISSUE TITLE
Joystick / Gamepad Nintendo Super

### ISSUE THREAD

**Issue by RafaelLinux** [Heroic-Games-Launcher/HeroicGamesLauncher#3209]
### Describe the bug

I'm using Heroic latest AppImage version for Linux, under Plasma/Wayland. Plasma shows my Nintendo Switch Pro Controller and let me calibrate it. I tested too with your [recommended](https://hardwaretester.com/gamepad) link in Firefox and it is detected as "057e-2009-Nintendo Switch Pro Controller" (see inserted screenshot).
![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/13048959/354eb657-1091-4c36-987e-85775cc90896)
However, "Heroic" doesn't react to my GamePad and games neither. I read in you documentation about "src/helper/gamepad.ts" file to add my GamePad, but I can't locate it. Where is that file and what should I copy inside it?

Thank you

### Add logs

```text
(19:58:55) INFO:    [Legendary]:        Legendary location: /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/legendary
(19:58:55) INFO:    [Gog]:              GOGDL location: /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/gogdl
(19:58:55) INFO:    [Connection]:       Connectivity: check-online
(19:58:55) INFO:    [Connection]:       Pinging external endpoints
(19:58:55) INFO:    [Nile]:             Refreshing library...
(19:58:55) INFO:    [Nile]:             Refreshing Amazon Games...
(19:58:55) INFO:    [Nile]:             Running command: NILE_CONFIG_PATH=/home/usuario/.config/heroic/nile_config /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/nile library sync
(19:58:55) INFO:    [Nile]:             Game list updated, got 20 games
(19:58:55) INFO:    [Backend]:          DRM module staus {
  "oimompecagnajdejgnnjijobebaeigek": {
    "name": "Widevine Content Decryption Module",
    "status": "new",
    "version": "4.10.2710.0"
  }
}
(19:58:57) WARNING: [Backend]:          Failed to register protocol with OS.
(19:58:57) INFO:    [Connection]:       Connectivity: online
(19:58:57) INFO:    [Gog]:              Checking if login is valid
(19:58:57) INFO:    [Gog]:              Running command: /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/usuario/.config/heroic/gog_store/auth.json auth
(19:58:57) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/usuario/.config/heroic/legendaryConfig/legendary /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/legendary --version
(19:58:57) INFO:    [Gog]:              Running command: /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/usuario/.config/heroic/gog_store/auth.json --version
(19:58:57) INFO:    [Nile]:             Running command: NILE_CONFIG_PATH=/home/usuario/.config/heroic/nile_config /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/nile --version
(19:58:57) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(19:58:57) INFO:    [Backend]:          Loading Screen Ready
(19:58:57) INFO:    [Backend]:          
System Information:
CPU: 24x AMD Ryzen 9 3900X 12-Core Processor
Memory: 33.55 GB (used: 18.87 GB)
GPUs:
  GPU 0:
    Name: Advanced Micro Devices, Inc. [AMD/ATI] Navi 33 [Radeon RX 7700S/7600/7600S/7600M XT/PRO W7600]
    IDs: D=7480 V=1002 SD=7601 SV=1eae
    Driver: amdgpu
OS: openSUSE Tumbleweed 6.5.9-1-default (linux)

The current system is not a Steam Deck
We are not running inside a Flatpak container

Software Versions:
  Heroic: 2.10.0 Jewelry Bonney
  Legendary: 0.20.33 Undue Alarm
  gogdl: 0.7.3
  Nile: 1.0.0 Jonathan Joestar

(19:58:58) INFO:    [Nile]:             Saved user data to config file
(19:58:58) INFO:    [Backend]:          Checking for new Heroic Updates
(19:58:58) INFO:    [Gog]:              Saved username to config file
(19:58:58) INFO:    [Frontend]:         Refreshing undefined Library
(19:58:58) WARNING: [Backend]:          refresh not implemented on Sideload Library Manager
(19:58:58) INFO:    [Gog]:              Running command: /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/usuario/.config/heroic/gog_store/auth.json auth
(19:58:58) INFO:    [Legendary]:        Refreshing library...
(19:58:58) INFO:    [Legendary]:        Refreshing Epic Games...
(19:58:58) INFO:    [Nile]:             Refreshing library...
(19:58:58) INFO:    [Nile]:             Refreshing Amazon Games...
(19:58:58) INFO:    [Nile]:             Running command: NILE_CONFIG_PATH=/home/usuario/.config/heroic/nile_config /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/nile library sync
(19:58:58) INFO:    [Nile]:             Game list updated, got 20 games
(19:58:58) INFO:    [Legendary]:        Game list updated, got 60 games & DLCs
(19:58:58) INFO:    [Backend]:          Frontend Ready
(19:58:58) INFO:    [Backend]:          Checking for current version changelog
(19:58:59) INFO:    [Winetricks]:       Downloading Winetricks
(19:58:59) INFO:    [Gog]:              Getting GOG library
(19:58:59) INFO:    [Gog]:              Running command: /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/usuario/.config/heroic/gog_store/auth.json auth
(19:58:59) INFO:    [Legendary]:        Running command: LEGENDARY_CONFIG_PATH=/home/usuario/.config/heroic/legendaryConfig/legendary /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(19:59:00) INFO:    [Gog]:              Saved games data
(19:59:00) WARNING: [Backend]:          listUpdateableGames not implemented on Sideload Library Manager
(19:59:00) INFO:    [Gog]:              Found 0 game(s) to update
(19:59:00) INFO:    [Legendary]:        Checking for game updates: LEGENDARY_CONFIG_PATH=/home/usuario/.config/heroic/legendaryConfig/legendary /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(19:59:00) INFO:    [Legendary]:        Found 0 games to update
(19:59:00) INFO:    [Nile]:             Looking for updates...
(19:59:00) INFO:    [Nile]:             Running command: NILE_CONFIG_PATH=/home/usuario/.config/heroic/nile_config /tmp/.mount_HeroicfSyFGv/resources/app.asar.unpacked/build/bin/linux/nile list-updates --json
(19:59:00) INFO:    [WineDownloader]:   Updating wine versions info
(19:59:00) INFO:    [WineDownloader]:   Fetching upstream information...
(19:59:01) INFO:    [WineDownloader]:   wine versions updated
(19:59:03) INFO:    [Backend]:          Starting the Download Queue
```

### Steps to reproduce

1. Connect GamePad
2. Launch "Heroic.AppImage"
3. Go to "Library" 
4. Try all joysticks from GamePad to move thru library items

### Expected behavior

To move thru Library with GamePad

### Heroic Version

Latest Stable

### System Information

CPU: 24x AMD Ryzen 9 3900X 12-Core Processor
Memory: 33.55 GB (used: 17.71 GB)
GPUs:
  GPU 0:
    Name: Advanced Micro Devices, Inc. [AMD/ATI] Navi 33 [Radeon RX 7700S/7600/7600S/7600M XT/PRO W7600]
    IDs: D=7480 V=1002 SD=7601 SV=1eae
    Driver: amdgpu
OS: openSUSE Tumbleweed 6.5.9-1-default (linux)

The current system is not a Steam Deck
We are not running inside a Flatpak container

Software Versions:
  Heroic: 2.10.0 Jewelry Bonney
  Legendary: 0.20.33 Undue Alarm
  gogdl: 0.7.3
  Nile: 1.0.0 Jonathan Joestar

**Comment by arielj** [comment]
can you try this?

- open heroic
- right click on the task icon > debug (this will open the developer tools)
- make sure you are in the `Console` tab and type `location = "https://hardwaretester.com/gamepad"` (this will open the gamepad tester url, but inside heroic, to see if electron actually has access to your controller)

if the controller works there, please share how the controller is identified inside (if it's different inside the electorn app) and what activates in that page for each button press and stick

the gamepad.ts file is meant for developers, not for users to modify it

**Comment by RafaelLinux** [comment]
I'm sorry but .... where is "task" icon? I can not locate it ....
![image](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/assets/13048959/3ad0c852-3564-42a9-8ca8-58c37d40f8c1)

**Comment by arielj** [comment]
I mean the icon that appears next to the system's clock (like volume, wifi, battery, etc)

**Comment by RafaelLinux** [comment]
I understood now. You mean the "Heroic" icon in taskbar ;)  ... done!!! But bad news ... same website doesn't detect my GamePad at all.

**Comment by RafaelLinux** [comment]
Update and correction: Only FIREFOX detect GamePad. Any other browser (Chromium, Chrome, Opera, MSEdge for Linux) doesn't detect any movement or button of my gamepad.

**Comment by arielj** [comment]
hmmm then we can't do much, heroic is an electron app using chromium and we depend on the browser reporting the controller to us

**Comment by RafaelLinux** [comment]
Ooops ... Electron applications are always a headache for me. I dropped Atom (made with Electron too) cause it showed "Archive" menu (when unfolded) in my left monitor, instead my main monitor, where Atom was fullscreen.

I'll try to find out about it out there ... I find something, I'll put it here.

Thank you

**Comment by arielj** [comment]
Closing this since there's nothing we can do in Heroic. Once chrome/electron fixes that support it should just work in Heroic.

**Comment by RafaelLinux** [comment]
Thank you anyway. I got a way (adding rules to system config) to make my joystick available for Heroic. Since the, Heroic is AWESOME now that I can use it w/o more issues.

Thank you!!

**Comment by arielj** [comment]
great! would be nice if you can share how you made it work in case someone has the same controller and the same issue and end up here

**Comment by RafaelLinux** [comment]
Of course.
My gamepad is a EasySMX. It's detected as "Nintendo Switch Pro Controller" in Linux. However, in Windows is recognized like a "Microsoft X-Box 360"  :open_mouth: 
I, accidentally, found out that it's a “**Thunderobot G30**”. A lot of retailers sell that gamepad with different brands and names. Thanks to that, I found a webpage that explained a trick to make it be recognized by my system, and I simplified it to a one-line command to execute from console (it will ask for root password):

`sudo sh -c "echo 'blacklist hid_nintendo' > /etc/modprobe.d/nonintendo.conf"`

Then reboot and, voilá, you will have a "Microsoft X-Box 360" controller in ALL your applications.
